wolfcrypt/src/ecc.c

```diff
@@ -4037,12 +4037,20 @@ int wc_ecc_make_key_ex(WC_RNG* rng, int keysize, ecc_key* key, int curve_id)
     CRYS_ECPKI_KG_TempData_t    tempBuff;
     CRYS_ECPKI_KG_FipsContext_t fipsCtx;
     byte ucompressed_key[ECC_MAX_CRYPTO_HW_SIZE*2 + 1];
-    word32 raw_size = (word32) (key->dp->size)*2 + 1;
+    word32 raw_size;
 #endif
     if (key == NULL || rng == NULL) {
         return BAD_FUNC_ARG;
     }
 
+#if defined(WOLFSSL_CRYPTOCELL)
+    if (key->dp == NULL) {
+        WOLFSSL_MSG("ECC internal dp structure was NULL");
+        return BAD_FUNC_ARG;
+    }
+
+    raw_size = (word32) (key->dp->size)*2 + 1;
+#endif
```

Comment:
Remove the block. 
Key->dp is set in the wc_ecc_reset() below it.

Reply:
changed